tools/read-dependencies

```diff
@@ -222,15 +229,23 @@ def main(distro):
     translated_pip_names = translate_pip_to_system_pkg(
         pip_pkg_names, renames)
     all_deps = []
+    select_requires = [args.build_requires, args.runtime_requires]
     if args.distro:
-        all_deps.extend(
-            translated_pip_names + deps_from_json['requires'] +
-            deps_from_json['build-requires'])
+        if all(select_requires) or not any(select_requires):
+            all_deps.extend(
+                translated_pip_names + deps_from_json['requires'] +
+                deps_from_json['build-requires'])
+        else:
+            if args.build_requires:
+                all_deps.extend(deps_from_json['build-requires'])
+            else:
+                all_deps.extend(deps_from_json['requires'])
```

Comment:
Hrm, I did not think requirements were a runtime requirement, but I guess they are.  It's awkward that we have packages/pkg-deps.json which has *some* but not all and requirements.txt which have other requirements, but not *all*; and in brpm, we call the requirements.txt under the test-deps ... 

Let me fix.  I'd like to refactor read-requirements a bit more, but that can happen in a follow up.

(1) python runtime requirements
(2) non-python runtime requirements 
(3) test runtime requirements
(4) build requirements

for packages, we only have two buckets, runtime and buildtime

So, for packaging, --runtime-requirements should dump (1) + (2)  (optionally (3) if build-time tests are enabled)
And --build-requirements will only dump (4)

Users of read-requirements may want to independently list 1, 2, 3 or 4.

Reply:
It's a bit messier then this, because it's not only a matter of which dependencies to list, but also *in which form* they are to be listed.

This is asking for (1):

```
$ tools/read-dependencies 
configobj
jinja2
jsonpatch
jsonschema
oauthlib
pyyaml
requests
```

while this is asking for (1) and (2) but with (1) translated as distro packages:

```
$ tools/read-dependencies -d debian -R
procps
python3-configobj
python3-jinja2
python3-jsonpatch
python3-jsonschema
python3-oauthlib
python3-requests
```